databricks/koalas/base.py

```diff
@@ -299,10 +301,25 @@ def is_monotonic(self):
 
         >>> ser.rename("a").to_frame().set_index("a").index.is_monotonic
         True
+
+        >>> ser = ks.Series([5, 4, 3, 2, 1], index=[1, 2, 3, 4, 5])
+        >>> ser.is_monotonic
+        False
+
+        >>> ser.index.is_monotonic
+        True
         """
+        from databricks.koalas.frame import DataFrame
+        from databricks.koalas.series import _col
         col = self._scol
-        window = Window.orderBy(self._kdf._internal.index_scols).rowsBetween(-1, -1)
-        return self._with_new_scol((col >= F.lag(col, 1).over(window)) & col.isNotNull()).all()
+        idx_tmp = SPARK_INDEX_NAME_FORMAT('temp')
+        sdf = self._internal.sdf.withColumn(idx_tmp, monotonically_increasing_id())
+        window = Window.orderBy(sdf[idx_tmp]).rowsBetween(-1, -1)
+        sdf = sdf.select(
+            sdf[idx_tmp],
+            F.when((col >= F.lag(col, 1).over(window)) & col.isNotNull(), True)
+             .otherwise(False).alias('__temp__'))
+        return sdf.where(F.col('__temp__') == 'false').count() < 2
```

Comment:
?? why does this compare to a string `false`?

Reply:
when i `show()` of the spark data frame, it looks string of 'false' in there, but  actually it is boolean internally, my mistake.